### PR TITLE
Overhaul transient buffer

### PR DESCRIPTION
**Pull request by quicknir**
Note that this PR includes my previous work, which has already been reviewed and approved. I make use of the new functions to add some nice bindings to buffer transient mode, making it useful for quickly rearranging many buffers.

When it's approved, as long as the other PR is merged before this one, the rebase should handle the duplicate commit.

### CODE REVIEW

**Comment by syl20bnr** [comment]
You won't need to rebase because we cherry-pick instead of merging, so it is on our side to cherry-pick only the second commit and not the first one ;-)

**Comment by quicknir** [comment]
@syl20bnr @d12frosted Hey guys, sorry to bother, but ping? I think this is a nice enhancement and I've been waiting for a review for a bit.

File: layers/+distributions/spacemacs-base/funcs.el [hunk 1]
```diff
@@ -184,6 +184,62 @@ Dedicated (locked) windows are left untouched."
   (interactive "p")
   (spacemacs/rotate-windows (* -1 count)))
 
+;; Note that this duplicates code from select-window-by-number, ideally should
+;; upstream this function into windows.el
+(defun spacemacs/get-window-by-number (i)
+  (let ((windows (car (gethash (selected-frame) window-numbering-table)))
```
Comment (d12frosted) [inline layers/+distributions/spacemacs-base/funcs.el]: Where is `window-numbering-table` coming from? It's not defined for me 😢

Comment (d12frosted) [inline layers/+distributions/spacemacs-base/funcs.el]: And it seems that this file already contains the same definition of `spacemacs/get-window-by-number`.

**Comment by quicknir** [comment]
@d12frosted Sorry for any confusion, I had pushed from this branch because the first commit contained necessary work for the second. When the first was merged, I offered to rebase this work to only have the second commit, but @syl20bnr wrote that it was not necessary in the comment above because the workflow is to cherry pick anyhow.

Would it be helpful to you if I just rebased anyhow?

**Comment by d12frosted** [comment]
@quicknir oh, it's my fault, I didn't read it carefully what you have written in your message. I was more focused on the code itself. 

> Would it be helpful to you if I just rebased anyhow?

As Sylvain said - since we cherry-pick it's not needed. 

P. S. thanks for clarifications :)

**Comment by d12frosted** [comment]
Thank you :+1: Cherry-picked into `develop`, you can safely delete your branch.

This works very nice 😸

**Pull request closed without merging**